vllm/model_executor/models/chatglm.py

```diff
@@ -306,26 +442,67 @@ def __init__(
                                            config.hidden_size,
                                            quant_config=quant_config)
 
+        vision_config_flag = getattr(config, 'vision_config', None)
+        if vision_config_flag is not None:
+            self.vision_config = Namespace(**config.vision_config)
+            self.vision = EVA2CLIPModel(self.config, quant_config)
+        else:
+            self.vision = None
+
     def forward(
         self,
         input_ids: torch.Tensor,
-        position_ids: torch.Tensor,
+        positions: torch.Tensor,
         kv_caches: List[torch.Tensor],
         attn_metadata: AttentionMetadata,
+        intermediate_tensors: Optional[IntermediateTensors] = None,
+        **kwargs: object,
     ) -> torch.Tensor:
+
         inputs_embeds = self.embedding(input_ids)
+        pixel_values = kwargs.pop("image_embeds", None)
```

Comment:
I assume `kwargs["image_embeds"]` are expected to be normalized images, right? If you write a custom mapper, can you map it to `pixel_values`? It's a bit confusing otherwise since a lot of the vision models support passing in the visual embeddings directly

Reply:
Ok, I will modify it in the new commit